bin/cython-generate-lexicon.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+
+#
+#   Updates Cython's Lexicon.py with the unicode characters that are accepted as
+#   identifiers. Should be run with the most recent version of Python possible
+#   to ensure that Lexicon is as complete as possible.
+#
+#   Python3 only (it relies on str.isidentifier which is a Python 3 addition)
+#
+#   Run with either
+#    --overwrite    to update the existing Lexicon.py file
+#    --here         to create a copy of Lexicon.py in the current directory
+
+import sys
+from io import StringIO
+import os
+
+# Make sure we import the right Cython
+cythonpath, _ = os.path.split(os.path.realpath(__file__)) # bin directory
+cythonpath, _ = os.path.split(cythonpath)
+if os.path.exists(os.path.join(cythonpath,"Cython")):
+    sys.path.insert(0, cythonpath)
+    print("Found (and using) local cython directory")
+# else we aren't in a development directory
+
+from Cython.Compiler import Lexicon
+
+def main():
+    if (len(sys.argv) != 2 or
+        (sys.argv[1] not in ['--overwrite','--here'])):
+        print("""Call the script with either:
+  --overwrite    to update the existing Lexicon.py file
+  --here         to create an version of Lexicon.py in the current directory
+""")
+        return
+
+    generated_code = StringIO()
+    print("# generated with:\n #", sys.implementation.name, sys.version, file=generated_code)
+    print(file=generated_code)
+    print(start_expression(), file=generated_code)
+    print(file=generated_code)
+    print(cont_expression(),file=generated_code)
+    print(file=generated_code)
+    generated_code = generated_code.getvalue()
+
+    output = StringIO()
+    mode = 0 # 1 when found generated section, 2 afterwards
+    print("Reading file", Lexicon.__file__)
+    with open(Lexicon.__file__,'r') as f:
+        for line in f:
+            if mode != 1:
+                output.write(line)
+            else:
+                if line.strip() == "# END GENERATED CODE":
+                    mode = 2
+                    output.write(line)
+            if mode == 0:
+                if line.strip() == "# BEGIN GENERATED CODE":
+                    mode = 1
+                    output.write(generated_code)
+
+    if mode != 2:
+        print("Warning: generated code section not found - code not inserted")
+        return
+
+    if sys.argv[1] == "--here":
+        outfile = "Lexicon.py"
+    elif sys.argv[1] == "--overwrite":
+        outfile = Lexicon.__file__
+    else:
+        raise ValueError('argv: "{0}" not recognised', sys.argv[1])
+
+    print("Writing to file", outfile)
+    with open(outfile,'w') as f:
+        f.write(output.getvalue())
+
+
+# The easiest way to generate an appropriate character set is just to use the str.isidentifier method
+# An alternative approach for getting character sets is at https://stackoverflow.com/a/49332214/4657412
+def get_start_characters_as_number():
```

Comment:
Feel free to use `@functools.lru_cache()` to avoid creating this list more than once.